src/miner.cpp

```diff
@@ -209,7 +209,7 @@ bool BlockAssembler::TestPackage(uint64_t packageSize, int64_t packageSigOpsCost
 //   segwit activation)
 bool BlockAssembler::TestPackageTransactions(const CTxMemPool::setEntries& package)
 {
-    for (const CTxMemPool::txiter it : package) {
+    for (CTxMemPool::txiter it : package) {
```

Comment:
I'd slightly prefer if you added a

```cpp
using const_txiter = indexed_transaction_set::nth_index<0>::type::const_iterator;
```
 and used that instead, since that seemed to be the intention here.

Reply:
Agreed!

Reply:
Actually, iterators are always const for multindex sets, so I'd slightly prefer if that was clarified in the header file:

```diff
diff --git a/src/txmempool.h b/src/txmempool.h
index bda812b42f..ebfcf36e11 100644
--- a/src/txmempool.h
+++ b/src/txmempool.h
@@ -486,7 +486,7 @@ public:
     mutable CCriticalSection cs;
     indexed_transaction_set mapTx GUARDED_BY(cs);
 
-    typedef indexed_transaction_set::nth_index<0>::type::iterator txiter;
+    using txiter = indexed_transaction_set::nth_index<0>::type::const_iterator;
     std::vector<std::pair<uint256, txiter> > vTxHashes; //!< All tx witness hashes/entries in mapTx, in random order
 
     struct CompareIteratorByHash {

Reply:
That can be done in a separate pull, though. Has nothing to do with your changes.